src/components/ParamsSelection/ParamsSelection.tsx

```diff
@@ -0,0 +1,174 @@
+import { FC, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { RootState } from '@/app/store';
+
+import { useCart } from '@/hooks';
+import { toggleClickedBuy } from '@/features/сart/сartSlice';
+import { getSeparetedCapacity, replaceParamsInPathname } from '@/utils';
+import { COLORS } from '@/constants/colors';
+import { PrimaryButtons } from '@/enums';
+import { Device, Product } from '@/types';
+
+import styles from './styles.module.scss';
+import {
+  ParameterButton,
+  PrimaryButton,
+  FavoritesButton,
+  ColorButton,
+} from '../index';
```

Comment:
When using this approach to imports, you should import from a shared index file only outside of the shared directory (in our case, outside of components). If the file is needed inside the directory, it should be imported from the component's folder. Otherwise it may cause [Circular dependency](https://en.wikipedia.org/wiki/Circular_dependency).
Because it turns out that to create a component, you import something from the file in which you export that same component, which is not the right approach